momepy/functional/_diversity.py

```diff
@@ -48,72 +138,103 @@ def describe(
-        if include_mode:
-            stat_["mode"] = grouper.agg(lambda x: _mode(x.values))
+    return _compute_stats(grouper, q, include_mode)
+
+
+def describe_reached(
+    y, y_group, result_index=None, graph=None, q=None, include_mode=False
+) -> DataFrame:
+    """
+    Calculates statistics of ``y`` objects reached on a street network.
+    Requires a ``y_group`` that links the ``y`` objects to streets (or ``graph``)
+    assigned beforehand (e.g. using :py:func:`momepy.get_network_id`).
+    The number of elements within neighbourhood defined in ``graph``. If
+    ``graph`` is ``None``, it will assume topological distance ``0`` (element itself)
+    and ``result_index`` is required in order to arrange the results.
+    If ``graph``, the results are arranged according to the spatial weights ordering.
+    The statistics calculated are count, sum, mean, median, std.
+    Optionally, mode can be calculated, or the statistics can be calculated in
+    quantiles ``q``.
+
+    Adapted from :cite:`hermosilla2012` and :cite:`feliciotti2018`.
+
+    Parameters
+    ----------
+    y : DataFrame | Series | numpy.array
+        A GeoDataFrame containing objects to analyse.
+    y_group : Series | numpy.array
+        The unique ID that specifies the aggregation
+        of ``y`` objects to ``graph`` groups.
+    result_index : pd.Index (default None)
+        An index that specifies how to order the results.
+    graph : libpysal.graph.Graph (default None)
+        A spatial weights matrix of the streets.
+    q : tuple[float, float] | None, optional
+        Tuple of percentages for the percentiles to compute. Values must be between 0
+        and 100 inclusive. When set, values below and above the percentiles will be
+        discarded before computation of the average. The percentiles are computed for
+        each neighborhood. By default None.
+    include_mode : False
+        Compute mode along with other statistics. Default is False. Mode is
+        computationally expensive and not useful for continous variables.
+
+    Returns
+    -------
+    DataFrame
+    """
+
+    if (result_index is None) and (graph is None):
+        raise ValueError(
+            "One of result_index or graph has to be specified, but not both."
+        )
+    elif result_index is None:
+        result_index = graph.unique_ids
+    elif graph is None:
+        result_index = result_index
     else:
-        agg = grouper.agg(lambda x: _describe(x.values, q=q, include_mode=include_mode))
-        stat_ = DataFrame(zip(*agg, strict=True)).T
-        cols = ["mean", "median", "std", "min", "max", "sum"]
-        if include_mode:
-            cols.append("mode")
-        stat_.columns = cols
+        raise ValueError(
+            "One of result_index or graph has to be specified, but not both."
+        )
 
-    return stat_
+    if isinstance(y, np.ndarray):
+        y = pd.Series(y, name="obs_index")
+    elif isinstance(y, Series):
```

Comment:
```suggestion
    if isinstance(y, Series):
```
to treat the series created above, no?